doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,605 @@
+This policy information can also be requested directly from remote ASes over gRPC.
+
+The control service introduces a FABRID service with the following endpoints where *intra-AS* means it can be reached
+from the local AS and *inter-AS* means it can be reached from a remote AS:
+
+- GetMPLSMapIfNecessary (intra-AS)
+    Is used by the router to retrieve the MPLS map for the intra-AS paths.
+    The map is only returned if the router does not have an up to date MPLS map.
+- GetRemotePolicyDescription (intra-AS)
+    Is used by the endhosts of the local AS to request the policy description of a policy identifier for a remote AS.
+- GetSupportedIndicesMap (inter-AS, intra-AS)
+    Returns the per interface-pair supported FABRID indices.
+- GetIndexIdentifierMap (inter-AS, intra-AS)
+    Returns a map that maps identifiers to indicies which can then be used for sending FABRID packets.
+- GetLocalPolicyDescription (inter-AS, intra-AS)
+    Is used to request the policy description of a policy identifier for the local AS.
+
+Important data structures
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following list explains the most important data structures used in the FABRID service:
+
+- SupportedIndicesMap
+    .. code-block:: go
+
+        type ConnectionPoint struct {
+            Type        string
+            IP          string
+            Prefix      uint32
+            InterfaceID uint16
+        }
+
+    Maps a connection pair consisting of two ConnectionPoints to a list of policy indices.
+    This map shows for each connection pair which policy indices are supported, which can be one or multiple policies.
+    A ConnectionPoint is either an interface, an IP range or wildcard.
+    For all intermediary hops interface to interface connection points will be used whereas interface to IP range is used for the last hop.
+- IndexIdentifierMap
+    A policy index is to be embedded in the HBH extension and therefore has to be minimal in size.
+    The size of a policy index is 8 bits, whereas identifiers can be a multiple of this (especially global identifiers).
+    The policy index is thus different to the policy identifier. In order to decode which policies are supported on which interfaces,
+    a mapping is required from policy index to local and global identifiers.
+    This mapping is provided by this map.
+- IdentifierDescriptionMap
+    Global identifiers can be found in a global datastore, but local identifiers are specific to an AS.
+    This map maps a local policy identifier to its corresponding description.
+- MPLSMaps
+    Routers need to be aware of the supported policy indices and the corresponding MPLS config they need to apply to packets to
+    enforce the policy in the internal network.
+    Routers periodically fetch this map from the control service.
+    A hash of the MPLS map is maintained, such that routers only have to update if their hash differs from the one at the control service.
+- RemotePolicyCache
```

Comment:
Is this also some kind of map?

If so, then all data structures listed here are maps -> we could then mention this at the beginning and would not have to use "Map" in the title of each map anymore, as it would be inherently clear.

A map title could maybe also look like: (ConnectionPoint, ConnectionPoint) -> []PolicyIndex
To make it specific.
What do you think?